lib/queue.js

```diff
@@ -45,8 +45,13 @@ class Queue extends Emitter {
       });
     }
 
-    this._delayedTimer = new EagerTimer(this.settings.nearTermWindow);
-    this._delayedTimer.on('trigger', this._activateDelayed.bind(this));
+    // To avoid changing the hidden class of the Queue.
+    this._delayedTimer = this.settings.activateDelayedJobs
```

Comment:
Yeah, it'd work. I prefer to keep properties on objects in all cases, because doing otherwise introduces a different hidden class. I don't expect that to really have any performance benefits. I also think it's nice to always have the full set of properties always present on an object, so you can see them easily in the REPL.